src/shared/cpu-set-util.c

```diff
@@ -287,3 +295,87 @@ int cpu_set_from_dbus(const uint8_t *bits, size_t size, CPUSet *set) {
         s = (CPUSet) {};
         return 0;
 }
+
+static int numa_policy_to_mempolicy(const NUMAPolicy *policy, unsigned long *maxnode, unsigned long **nodes) {
+        unsigned node, bits = 0, ulong_bits;
+        _cleanup_free_ unsigned long *out = NULL;
+
+        assert(policy);
+        assert(maxnode);
+        assert(nodes);
+
+        /* Take a shortcut here if policy is set to default or local since kernel requires that we pass NULL as nodemask */
+        if (IN_SET(policy->type, NUMA_POLICY_DEFAULT, NUMA_POLICY_LOCAL))
+                goto out;
```

Comment:
if this branch is taken then `*maxnode` will be 1 afterwards, even though *nodes is NULL. Does that make sense?